### ISSUE TITLE
Vite build fails on import analysis

### ISSUE THREAD

**Issue by orta** [lorefnon/vite-plugin-civet#3]
This happens with the blank template, perhaps there's a missing part of the vite plugin API?

```
> yarn vite build
yarn run v1.22.17
$ /home/orta/dev/civet/civet-blank/node_modules/.bin/vite build
vite v4.0.4 building for production...
✓ 2 modules transformed.
[vite:build-import-analysis] Parse error @:8:28
file: /home/orta/dev/civet/civet-blank/src/main.civet:8:27
 6:         <React.StrictMode>
 7:             <App />
 8:         </React.StrictMode>
                               ^
 9:     ),
10:     document.getElementById 'root'
error during build:
Error: Parse error @:8:28
    at parse$e (file:///home/orta/dev/civet/civet-blank/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:14710:355)
    at Object.transform (file:///home/orta/dev/civet/civet-blank/node_modules/vite/dist/node/chunks/dep-5e7f419b.js:43370:27)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by lorefnon** [comment]
I may not be able to look into this now (travelling for personal work), but looks like this problem only happens when a civet file is referenced from index.html. 

So you can work around this by having a typescript entry point (which itself can reference other civet files). The solid-playground template does this and the build works fine there.

**Comment by edemaine** [comment]
Actually, this isn't the issue. I changed the Solid example to use a `.civet` entry point and everything works fine there.

The issue is that the React transform isn't getting called after the Civet → JS transformation. This is because vite-plugin-react-swc [uses `config` instead of `transform` during build mode](https://github.com/vitejs/vite-plugin-react-swc/blob/786826fb81459d7e088703a048d5523f2841f7cd/src/index.ts#L129-L141), different from dev mode. There is a new `plugins` option to force it to use a `transform` (from 3 days ago), but it isn't released yet.

Ideally we'd pass through the `config` settings of vite-plugin-react-swc as our own `config`. But this is a little tricky to do, because it's easiest to find vite-plugin-react-swc during the `configResolved` event, but I think then it's too late to add to our config. Instead it seems like we might need to find the relevant vite-plugin-react-swc during the `config` event (filtering out the one with `config.apply === command`). Unless someone knows a better way?

**Comment by lorefnon** [comment]
Yeah I realized this after trying to update the react example with a ts entrypoint. 

Simplest workaround for now is to use the transformOutput config option to transform the output (through babel) instead of the outputTransformerPlugin. I have an added an example in playground-react-babel dir which does this.

**Comment by lorefnon** [comment]
Available in 0.2.5
